enterprise_gateway/services/processproxies/k8s.py

```diff
@@ -308,14 +312,61 @@ def _create_kernel_namespace(self, service_account_name: str) -> str:
                     body = client.V1DeleteOptions(
                         grace_period_seconds=0, propagation_policy="Background"
                     )
-                    client.CoreV1Api().delete_namespace(name=namespace, body=body)
+                    client.CoreV1Api(api_client=kubernetes_client).delete_namespace(name=namespace, body=body)
                     self.log.warning(f"Deleted kernel namespace: {namespace}")
                 else:
                     reason = f"Error occurred creating namespace '{namespace}': {err}"
                 self.log_and_raise(http_status_code=500, reason=reason)
 
         return namespace
 
+    def _create_service_account_if_not_exists(self, namespace: str, service_account_name: str) -> None:
+        service_account_list_in_namespace: client.V1ServiceAccountList = client\
+            .CoreV1Api(api_client=kubernetes_client)\
+            .list_namespaced_service_account(namespace=namespace)
+
+        service_accounts_in_namespace: List[client.V1ServiceAccount] = service_account_list_in_namespace.items
+        service_account_names_in_namespace: List[str] = list(map(lambda svcaccount: svcaccount.metadata.name, service_accounts_in_namespace))
+
+        if service_account_name not in service_account_names_in_namespace:
```

Comment:
1. Super super good on you for using functional programming here. This is a senior dev move.

2. We can make this an even more big-headed by using a [generator comprehension](https://www.pythonlikeyoumeanit.com/Module2_EssentialsOfPython/Generators_and_Comprehensions.html#Creating-your-own-generator:-generator-comprehensions). The list/map method is on the order of n in terms of memory complexity and roughly 2n in terms of time complexity (we build a list of length n which takes n time, then we scan the list which takes n time). The lambda also requires constructing a function context, which can be expensive. The generator method is on the order of o(1) in terms of memory complexity and takes only n in terms of time complexity (we just perform a single scan of the other list without actually constructing anything). 

```suggestion
        service_account_names_in_namespace = (svcaccount.metadata.name for svcaccount in service_accounts_in_namespace)
        
        if service_account_name not in service_account_names_in_namespace:
```

Reply:
At this point, you are probably thinking "why do we care about performance, this is not that performance intensive?" 

This has less to do with performance and more to do with code quality; we want the Juypter reviewers to think "damn, thats some slick code, they must know what they are doing."

Reply:
Oh I love finding out about optimizations like this, I honestly don't know enough about Python for how much I use it, so thanks for teaching me about this 😄

Reply:
Oh it's list comprehension... darn I should've remembered about that, I keep defaulting to the list-map combo, probably because I use .map a lot in TS